### PR TITLE
Docs: Edit the Tools and building section

### PR DESCRIPTION
**Pull request by kirszenbaum**
This PR edits the **Tools and building** section in line with the #9737 guidelines.

Updated pages:
- http://localhost:8080/docs/next/react-data-grid/modules/
- http://localhost:8080/docs/next/react-data-grid/custom-plugins
- http://localhost:8080/docs/next/react-data-grid/custom-builds/

IMO the following pages don't require any updates (React-wise), it's worth double-checking, though:
- http://localhost:8080/docs/next/react-data-grid/packages
- http://localhost:8080/docs/next/react-data-grid/testing
- http://localhost:8080/docs/next/react-data-grid/folder-structure/

[skip changelog]

### CODE REVIEW

File: docs/content/guides/tools-and-building/custom-plugins.md [hunk 1]
```diff
@@ -267,15 +271,50 @@ const hotInstance = new Handsontable(container, {
...
+<HotTable
+  // Pass `true` to enable the plugin with default options.
+  [CustomPlugin.PLUGIN_KEY]={true}
+  // You can also enable the plugin by passing an object with options.
+  [CustomPlugin.PLUGIN_KEY]={{
+    msg: 'user-defined message',
+  }}
+  // You can also initialize the plugin without enabling it at the beginning.
+  [CustomPlugin.PLUGIN_KEY]={false}
+/>
```
Comment (budnix) [inline docs/content/guides/tools-and-building/custom-plugins.md:290]: AFAIK JSX does not support computed properties. I've changed the snippet to use the plugin key the same as the example above the section.
```suggestion
<HotTable
  // Pass `true` to enable the plugin with default options.
  customPlugin={true}
  // You can also enable the plugin by passing an object with options.
  customPlugin={{
    msg: 'user-defined message',
  }}
  // You can also initialize the plugin without enabling it at the beginning.
  customPlugin={false}
/>
```

File: docs/content/guides/tools-and-building/modules.md [hunk 2]
```diff
@@ -166,9 +167,33 @@ new Handsontable(container, {
...
+
+const container = document.querySelector('#example1');
+
+<HotTable
+  columns={[
+    {
+    type: 'numeric'
+    }
+  ]}
+/>
```
Comment (budnix) [inline docs/content/guides/tools-and-building/modules.md:190]: Fix formatting :) 
```suggestion
<HotTable
  columns={[
    {
      type: 'numeric'
    }
  ]}
/>
```

**Review by budnix — CHANGES_REQUESTED** [review]

**Review by budnix — APPROVED** [review]
👍

**Comment by bulaj** [comment]
I think `packages` page from reacts point of view feels so out of this world that its almost pointless like this - don't think anybody will embed scripts on its page like this rather than using imports.
I'm not sure that we have a valid alternative for this page in reacts context. At most we could suggest something similar to 'Bare distribution' just using imports, but it doesn't feel particularly useful. And internalisation section that would be almost exact copy of what we have in languages section.

**Pull request merged**